src/Tribe/Views/V2/Messages.php

```diff
@@ -0,0 +1,248 @@
+			'no_results_found_w_keyword'       => __(
+				'There were no results found for <strong>"%1$s"</strong>.',
+				'the-events-calendar'
+			),
+			// translators: the placeholder is the keyword(s), as the user entered it in the bar.
+			'month_no_results_found_w_keyword' => __(
+				'There were no results found for <strong>"%1$s"</strong> this month. Try searching next month.',
+				'the-events-calendar'
+			),
+			// translators: the placeholder is the formatted date string, e.g. "February 22, 2020".
+			'day_no_results_found'             => __(
+				'No events scheduled for %s. Please try another day.',
+				'the-events-calendar'
+			),
+		];
+
+		/**
+		 * Filters the map of user-facing messages that will be used in the Views.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $map An map of message keys to localized, user-facing, messages.
+		 */
+		$map = apply_filters( 'tribe_events_views_v2_messages_map', $map );
+
+		// If not found return the key itself.
+		$match = Arr::get( $map, $key, $key );
+
+		return count( $values ) ? sprintf( $match, ...$values ) : $match;
+	}
+
+	/**
+	 * Applies the current message render policy to the messages and returns an array of messages.
+	 *
+	 * @since TBD
+	 *
+	 * @return array An array of messages in the shape `[ <message_type> => [ ...<messages> ] ]`.
+	 */
+	public function to_array() {
+		return $this->apply_render_strategy( $this->messages );
+	}
+
+	/**
+	 * Applies the render strategy to the collection of messages.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $messages The collection of messages to apply the render strategy to.
+	 *
+	 * @return array An array of messages after the current strategy application.
+	 *               No matter the render strategy, the array always has shape
+	 *              `[ <message_type> => [ ...<messages> ] ]`.
+	 */
+	protected function apply_render_strategy( array $messages = [] ) {
+		if ( empty( $messages ) ) {
+			return [];
+		}
+
+		$updated_messages = $this->messages;
+
+		switch ( $this->render_strategy ) {
+			case static::RENDER_STRATEGY_PRIORITY_LAST:
+				array_walk(
+					$updated_messages,
+					static function ( array &$value, $message_type ) {
+						ksort( $value );
+						// Keep the highest priority (lower number).
+						$highest = (array) reset( $value );
+						// Keep only the last message.
+						$value = [ end( $highest ) ];
+					}
+				);
+				break;
+			case static::RENDER_STRATEGY_PRIORITY_FIRST:
+				array_walk(
+					$updated_messages,
+					static function ( array &$value, $message_type ) {
+						ksort( $value );
+						// Keep the highest priority (lower number).
+						$highest = (array) reset( $value );
+						// Keep only the first message.
+						$value = [ reset( $highest ) ];
+					}
+				);
+				break;
+			default:
+			case static::RENDER_STRATEGY_LIST:
+			array_walk(
+				$updated_messages,
+				static function ( array &$value, $message_type ) {
+					ksort( $value );
+				}
+			);
+			break;
+		}
```

Comment:
:no_entry_sign: Terminating statement must be indented to the same level as the CASE body

<sub>hash: 2069a48656281a4ecda8b60d2930d11d</sub>